### PR TITLE
Allow instructor assistant edit parent instructors posts

### LINKED ISSUE #2122: Instructor's Assistant Permissions
### Reproduction Steps

+ This isn't connected to a ticket I just found this while looking into the other instructor permissions. 
+ Create an instructor's assistant user
+ Assign them to an instructor who has created a course and is the author
+ Switch into that user

### Expected Behavior

+ they can edit courses that are authored by the instructors they’ve been assigned to

### Actual Behavior

+ They cannot edit the courses _unless_ they are assigned as an instructor.
+ However, if they are assigned as an instructor they have the ability to add/modify the access plans

### Error Messages / Logs

+ Include any relevant error messages or log files
```
<!-- Paste error logs / backtraces below this line -->

```

### System and Environment Information

<details>
<summary>System Report</summary>

```
Wordpress
-------------------------------------------

Home Url: https://natalie-test-site.myliftersite.com
Site Url: https://natalie-test-site.myliftersite.com
Login Url: https://natalie-test-site.myliftersite.com/wp-login.php
Version: 5.9.3
Debug Mode: No
Debug Log: No
Debug Display: Yes
Locale: en_US
Multisite: No
Page For Posts: Not Set
Page On Front: Homepage (#1398) [https://natalie-test-site.myliftersite.com/]
Permalink Structure: /%postname%/
Show On Front: page
Wp Cron: Yes


Settings
-------------------------------------------

Version: 6.3.0
Db Version: 6.3.0
Course Catalog: Course Catalog (#35) [https://natalie-test-site.myliftersite.com/courses/]
Membership Catalog: Membership Catalog (#36) [https://natalie-test-site.myliftersite.com/memberships/]
Student Dashboard: Dashboard (#38) [https://natalie-test-site.myliftersite.com/dashboard/]
Checkout Page: Checkout (#715) [https://natalie-test-site.myliftersite.com/checkout/]
Course Catalog Per Page: 9
Course Catalog Sorting: menu_order,ASC
Membership Catalog Per Page: 9
Membership Catalog Sorting: menu_order
Site Membership: Not Set
Courses Endpoint: my-courses
Edit Endpoint: edit-account
Lost Password Endpoint: lost-password
Vouchers Endpoint: redeem-voucher
Autogenerate Username: no
Password Strength Meter: no
Minimum Password Strength: 
Terms Required: yes
Terms Page: Activate (#443) [https://natalie-test-site.myliftersite.com/activate/]
Checkout Names: 
Checkout Address: 
Checkout Phone: 
Checkout Email Confirmation: no
Open Registration: yes
Registration Names: 
Registration Address: 
Registration Phone: 
Registration Voucher: 
Registration Email Confirmation: no
Account Names: 
Account Address: 
Account Phone: 
Account Email Confirmation: no
Confirmation Endpoint: confirm-payment
Force Ssl Checkout: no
Country: IT
Currency: USD
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Decimals: 2
Trim Zero Decimals: no
Recurring Payments: yes
Email From Address: team+sandbox@lifterlms.com
Email From Name: A LifterLMS Sandbox
Email Footer Text: 
Email Header Image: 
Cert Bg Width: 1123
Cert Bg Height: 794
Cert Legacy Compat: no


Constants
-------------------------------------------

LLMS_REMOVE_ALL_DATA: undefined
LLMS_REST_DISABLE: undefined
LLMS_SITE_FEATURE_RECURRING_PAYMENTS: undefined
LLMS_SITE_IS_CLONE: undefined


Gateways
-------------------------------------------

Manual: Enabled
Manual Logging: no
Manual Order: 1


Server
-------------------------------------------

Mysql Version: 5.7.37
Php Curl: Yes
Php Default Timezone: UTC
Php Fsockopen: Yes
Php Max Input Vars: 5000
Php Max Upload Size: 512 MB
Php Memory Limit: 256M
Php Post Max Size: 1024M
Php Soap: Yes
Php Suhosin: No
Php Time Limt: 30
Php Version: 8.0.17
Software: Apache/2.4.53 (Unix) OpenSSL/1.1.1
Wp Memory Limit: 256M


Browser
-------------------------------------------

HTTP USER AGENT: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.88 Safari/537.36


Theme
-------------------------------------------

Name: LifterLMS LaunchPad
Version: 2.4.9
Themeuri: https://lifterlms.com/launchpad/
Authoruri: https://gocodebox.com
Template: 
Child Theme: No
Llms Support: No


Plugins
-------------------------------------------

LifterLMS: 6.3.0
User Switching: 1.5.8


Integrations
-------------------------------------------

BbPress: No
BuddyPress: No


Template Overrides
-------------------------------------------




```

</details>

This issue has be recreated:
+ [x] Locally
+ [x] On a trial site
+ [ ] On a production website
+ [x] With only LifterLMS and a default theme

### Browser, Device, and Operating System Information

+ Browser name and version
+ Operating System name and version
+ Device name and version (if applicable)

### PR DESCRIPTION
**Pull request by eri-trabiccolo**
## Description

Fixes #2122 

## How has this been tested?
manually and new unit tests.

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: includes/models/model.llms.instructor.php [hunk 1]
```diff
@@ -125,35 +126,60 @@ public function get_memberships( $args = array(), $return = 'llms_posts' ) {
...
+	 * @param boolean $include_assistant Include posts whose instructor is just a parent of this instructor.
+	 * @return mixed
 	 */
-	public function get_posts( $args = array(), $return = 'llms_posts' ) {
+	public function get_posts( $args = array(), $return = 'llms_posts', $include_assistant = false ) {
 
-		$serialized_id = serialize( // phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.serialize_serialize
+		$parent_instructors = $include_assistant ? $this->get( 'parent_instructors' ) : array();
+		$instructor_ids     = array_merge(
+			$parent_instructors,
```
Comment (pondermatic) [inline includes/models/model.llms.instructor.php]: Reduce the number of lines it takes to add a single array element.
```suggestion
		$instructor_ids = $include_assistant ? $this->get( 'parent_instructors' ) : array();
		$instructor_ids[] = $this->get_id();
```
GitHub wouldn't let me suggest with all the lines used in `array_merge()`.

**Comment by pondermatic** [comment]
It looks like this will work @eri-trabiccolo.

File: includes/models/model.llms.instructor.php (hunk 1 above)
Comment (eri-trabiccolo) [reply includes/models/model.llms.instructor.php]: weird github.
Thanks for the suggestion!

**Comment by eri-trabiccolo** [comment]
I wonder if we should first fix #2121 prior to merging this, because as of now assistants who are not set as instructors of a course just cannot edit it, so they cannot even change/add access plans. But with this PR they will be able to do it as long as they can access the course edit UI.

**Comment by pondermatic** [comment]
@eri-trabiccolo wrote,
> I wonder if we should first fix #2121 prior to merging this,

Very good point. We don't want to release a new feature, only to take it away (unless a setting is changed) in the next release.

**Pull request closed without merging**